src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -177,7 +177,7 @@ object OpCode {
   * @param delta number of words to be popped from stack
   * @param alpha number of words to be pushed to stack
   */
-sealed abstract class OpCode(val code: Byte, val delta: Int, val alpha: Int, val constGasFn: FeeSchedule => BigInt) {
+abstract class OpCode(val code: Byte, val delta: Int, val alpha: Int, val constGasFn: FeeSchedule => BigInt) extends Product with Serializable {
```

Comment:
`VMSuite`, line 82:

```
if (!Set(TestCREATE, TestCALL, TestCALLCODE).contains(itx.opcode))
```

fails to compile without this. To me it looks like a bug in the type inferencer. If I change this line to:

```
if (!Set[OpCode](TestCREATE, TestCALL, TestCALLCODE).contains(itx.opcode))
```

then it compiles.

But adding `extends Product with Serializable` also works, and in practice it also means that we're requiring implementations of `OpCode` to be case objects/classes - which makes sense to do.